### ISSUE TITLE
Implement more loss functions

### ISSUE THREAD

**Issue by pesekon2** [ctu-geoforall-lab-projects/phd-pesek-2022#28]
It would be nice to compare results also for trainings done using different types of loss functions. It is also being stated that boundary-based losses or compound losses could be valuable for imbalanced data.

* distribution-based losses:
  - [x] cross entropy
* compound losses:
  - [ ] DiceFocal
* region-based losses:
  - [x] Dice
  - [x] Tversky
* boundary-based losses:
  - [ ] Hausdorff Distance

For more, exempli gratia: <https://jeune-research.tistory.com/entry/Loss-Functions-for-Image-Segmentation-Distance-Based-Losses>
